src/queries/getDependencies.ts

```diff
@@ -1,10 +1,23 @@
 import { asyncExec } from './exec';
 import { PackageList, Manager } from '../types';
 
+const AUTO_EXCLUDE = [
+  // automatically added to `node_modules` in Angular projects
+  '__ngcc_entry_points__.json',
+];
+
 export async function getDependencies(manager: Manager): Promise<PackageList> {
   try {
     const list = await asyncExec(manager.list);
-    return JSON.parse(list.stdout).dependencies;
+    const parsedList = JSON.parse(list.stdout).dependencies;
+    return Object.keys(parsedList)
+    .filter((dep) => !AUTO_EXCLUDE.includes(dep))
+    .reduce((acc, curr) => {
```

Comment:
Hey @kindoflew, we can decrease the amount of loops here implementing the filtering logic inside reduce, e.g.:
```javascript
(acc, next) => {
  if (AUTO_EXCLUDE.includes(dep)) {
    return acc;
  }
  return {
    ...acc,
    [curr]: parsedList[curr],
  } 
}
```

What do you think about this?

Reply:
oohhh good call! thanks for the tip, i'm still new to abusing `reduce` 😂

Reply:
heh, yeah, it is confusing to start with, but then, it may make life much easier)